test/metabase/query_processor/test_util.clj

```diff
@@ -276,6 +276,19 @@
     (1 1M true)  true
     (throw (ex-info "value is not boolish" {:value x}))))
 
+(defn format-nil->empty-str
+  "Return a function that converts nil to an empty string if driver treats empty strings a null.
+
+  Can be used with [[format-rows-by]] to normalize empty strings in results (but don't forget you need to pass
+  `format-nil-values?` when calling the [[format-rows-by]])."
+  [driver]
+  (if-not (driver.tu/empty-string-is-null? driver)
+    str
+    (fn nil->empty-string [value]
+      (if (nil? value)
+        ""
+        (str value)))))
```

Comment:
Since
```clojure
user> (str nil)
""
```
this is could be simplified to `(def format-nil->empty-str str)`, so I wouldn't even bother introducing it.

Reply:
Aha! I mistakenly assumed `(str nil)` was returning `nil` because I didn't know about the `format-nil-values?` arg to `format-rows-by`. Once I discovered it, it didn't occur to me that `format-nil-values?` was the whole explanation, and this function func is unnecessary. Good catch!